northd/northd.c

```diff
@@ -10773,12 +10773,26 @@ build_lrouter_nat_flows_for_lb(struct ovn_lb_vip *lb_vip,
         }
 
         if (lb_vip->n_backends || !lb_vip->empty_backend_rej) {
-            new_match_p = xasprintf("%s && is_chassis_resident(%s)",
-                                    new_match,
-                                    od->l3dgw_ports[0]->cr_port->json_key);
-            est_match_p = xasprintf("%s && is_chassis_resident(%s)",
-                                    est_match,
-                                    od->l3dgw_ports[0]->cr_port->json_key);
+            if (od->n_l3dgw_ports > 1) {
```

Comment:
What if the VIP is not in any of the LRP's subnets?

Reply:
Also, we probably need some tests to ensure this behavior doesn't change by accident in the future.

Reply:
Please also [sign-off on](https://github.com/ovn-org/ovn/blob/main/Documentation/internals/contributing/submitting-patches.rst#developers-certificate-of-origin) your commits so we can accept them in the repository.